### DISCUSSION TITLE
Run `machine-learning` on Python 3.12

### DISCUSSION THREAD

**Discussion by sorz** [immich-app/immich#10474] [General]
Python version is pined at 3.10 & 3.11 for now.

https://github.com/immich-app/immich/blob/517a83cfa9082ad9b7934a60dd8a5dc199e28004/machine-learning/pyproject.toml#L10

I wounder if there is any known compatibility issues that prevent `machine-learning` from running on Python 3.12, or just because it have not been test on that version yet?

The version limitation cause some difficulties on downstream Arch Linux packages, see comments on https://aur.archlinux.org/packages/immich-server
I'm looking for if there is anything I can do to improve it.

**Comment by mertalev** [comment]
It was to keep Renovate from trying to upgrade everything to 3.12 prematurely right when it released. 3.12 support is relatively mature now so I don’t see an issue with removing the constraint.
